exporter/ethstore.go

```diff
@@ -48,11 +48,22 @@ func StartEthStoreExporter(bnAddress string, enAddress string, updateInterval, e
 		ese.Sleep = time.Minute
 	}
 
+	// Reexport days if specified
+	if startDayReexport != -1 && endDayReexport != -1 {
+		for day := startDayReexport; day <= endDayReexport; day++ {
+			err := ese.exportDay(strconv.FormatInt(day, 10), true)
+			if err != nil {
+				logger.WithError(err).Errorf("error reexporting eth.store day %d in database", day)
+				return
+			}
+		}
+	}
+
 	ese.Run()
 
 }
 
-func (ese *EthStoreExporter) ExportDay(day string) error {
+func (ese *EthStoreExporter) exportDay(day string, shouldClearDay bool) error {
```

Comment:
**Suggestion:** Semantically, the day should not be cleared by a function called `exportDay`. There should maybe be a `clearDay` (or `clearDays`) function that is only called here: https://github.com/gobitfly/eth2-beaconchain-explorer/blob/4ad8da36a4a34dfa029705fc50f96a971f7dd85e/exporter/ethstore.go#L51-L54

Of course, you would have to think about the `tx` then.

That way, the normal "runner" would not have to mind the new bool and everything would be separated cleanly.

---

A different approach would be to add an `ON CONFLICT` clause to the `INSERT` queries. Out of curiosity: Why didn't you just do that?

Reply:
As you said my issue was that I wanted to keep everything for one day within a single `tx` so that we don't have a bad timing where a day is deleted for a short while.

I did not do an `ON CONFLICT` handling for the `eth_store_stats` because:
* First of for the `validator = -1` entry of the day there should not be a problem with it
* For the normal validator entries we have the problem that with the primary keys day & validator we do not delete obsolete entries, e.g.

before reexport:
day 10, validator 1, values: ...
day 10, validator 2, values: ...
day 10, validator 3, values: ...
day 10, validator 4, values: ...
day 10, validator 5, values: ...

after reexport it should be:
day 10, validator 1, values: ...
day 10, validator 3, values: ...
day 10, validator 5, values: ...

Then the values for validators 1, 3 and 5 get updated but 2 and 4 do not get deleted.
I don't know if it is possible for validators to become obsolete in that table but I did not want to risk it.

Reply:
I created new methods `prepareClearDayTx` & `prepareExportDayTx` which are called in `reexportDay` & `exportDay`.
Please check if that is the desired handling.